src/DDNSUpdate/Application/Providers/DigitalOcean/DigitalOceanClient.cs

```diff
@@ -26,18 +26,20 @@ public DigitalOceanClient(HttpClient httpClient)
             _httpClient = new FlurlClient(httpClient);
         }
 
-        public async Task<Result> CreateDNSRecordAsync(DigitalOceanCreateDomainRecordRequest request, string token, CancellationToken cancellation)
+        public async Task<Result> CreateDNSRecordAsync(string domainName, DigitalOceanCreateDomainRecordRequest request, string token, CancellationToken cancellation)
         {
-            string path = string.Format(_createDNSRecordFormat, request.Name);
-            string json = JsonConvert.SerializeObject(request);
-            HttpResponseMessage response = await BuildRequest(token, path).PostJsonAsync(json, cancellation);
-            return Result.OkIf(response.StatusCode == HttpStatusCode.Created, $"Unable to create DNS record for {request.Name}.");
+            string path = string.Format(_createDNSRecordFormat, domainName);
+            IFlurlRequest httpRequest = BuildRequest(token, path);
+            HttpResponseMessage response = await httpRequest.PostJsonAsync(request, cancellation);
+            bool created = (int)response.StatusCode == (int)HttpStatusCode.Created;
+            return Result.OkIf(created, $"Unable to create DNS record for {request.Name}.");
         }
 
         public async Task<Result<DigitalOceanGetDomainRecordsResponse>> GetDNSRecordsAsync(DigitalOceanDomain domain, string token, CancellationToken cancellation)
         {
             string path = string.Format(_getDNSRecordFormat, domain.Name);
-            HttpResponseMessage response = await BuildRequest(token, path).GetAsync(cancellation);
+            IFlurlRequest httpRequest = BuildRequest(token, path);
```

Comment:
Is there a reason this is split up? Could still be 1-lined

Reply:
It's confusing to return a HttpResponseMessage from BuildRequest.
I doubt people would notice GetAsync on the end.

Reply:
Eh sure